test/transpile/paths.spec.ts

```diff
@@ -0,0 +1,138 @@
+
+// Path for project tsconfig.json to resolve for
+const configFilePath = path.join(cwd, "tsconfig.json");
+
+describe("getSourceDir", () => {
+    test("with rootDir", () => {
+        const program = ts.createProgram(["main.ts", "src/otherfile.ts"], { configFilePath, rootDir: "src" });
+
+        // getCommonSourceDirectory does not work right so mock it
+        jest.spyOn(program, "getCommonSourceDirectory").mockReturnValue(cwd);
+
+        expect(getSourceDir(program)).toBe(path.join(cwd, "src"));
+    });
+
+    test("without rootDir", () => {
+        const program = ts.createProgram(["main.ts", "src/otherfile.ts"], { configFilePath });
+
+        // getCommonSourceDirectory does not work right so mock it
+        jest.spyOn(program, "getCommonSourceDirectory").mockReturnValue(cwd);
+
+        // Common sources directory is project root
+        expect(normalize(getSourceDir(program))).toBe(cwd);
+    });
+
+    test("without rootDir in src dir", () => {
+        const program = ts.createProgram([path.join(cwd, "src", "main.ts"), path.join(cwd, "src", "otherfile.ts")], {
+            configFilePath,
+        });
+
+        // getCommonSourceDirectory does not work right so mock it
+        jest.spyOn(program, "getCommonSourceDirectory").mockReturnValue(path.join(cwd, "src"));
+
+        // Common sources directory is src
+        expect(normalize(getSourceDir(program))).toBe(path.join(cwd, "src"));
+    });
+});
+
+describe("getEmitOutDir", () => {
+    test("with outDir", () => {
+        const program = ts.createProgram(["main.ts", "src/otherfile.ts"], { configFilePath, outDir: "out" });
+
+        // getCommonSourceDirectory does not work right so mock it
+        jest.spyOn(program, "getCommonSourceDirectory").mockReturnValue(cwd);
+
+        expect(getEmitOutDir(program)).toBe(path.join(cwd, "out"));
+    });
+
+    test("without outDir", () => {
+        const program = ts.createProgram(["src/main.ts", "src/otherfile.ts"], { configFilePath });
+
+        // getCommonSourceDirectory does not work right so mock it
+        jest.spyOn(program, "getCommonSourceDirectory").mockReturnValue(path.join(cwd, "src"));
+
+        expect(normalize(getEmitOutDir(program))).toBe(cwd);
+    });
+});
+
+describe("getEmitPath", () => {
+    test("puts files next to input without options", () => {
+        const { transpiledFiles } = util.testModule``
+            .setMainFileName("main.ts")
+            .addExtraFile("dir/extra.ts", "")
+            .expectToHaveNoDiagnostics()
+            .getLuaResult();
+
+        const fileNames = transpiledFiles.map(f => f.outPath);
+        expect(fileNames).toContain("main.lua");
+        expect(fileNames).toContain(path.join("dir", "extra.lua"));
+    });
+
+    test("puts files in outdir", () => {
+        const { transpiledFiles } = util.testModule``
+            .setMainFileName("main.ts")
+            .addExtraFile("dir/extra.ts", "")
+            .setOptions({ outDir: path.join(cwd, "tstl-out") })
```

Comment:
Maybe add a var for `path.join(cwd, "tstl-out")` that can be reused in the asserts below.